### PR TITLE
fix(app-shell): apollo header link to omit empty headers

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

This pull request omits empty header values. Instead of passing potentially `null` through the links we remove the value.

#### Description

This is an attempt to reduce the "Failed to fetch"-errors. On some issues on Apollo it was said that doing this or passing `''` removed the issue. I think it might be worth a shot.

### CODE REVIEW

File: packages/application-shell/src/apollo-links/header-link.js [hunk 1]
```diff
@@ -7,7 +8,9 @@ import {
   selectUserId,
 } from '../utils';
 
-const isKnownTarget = target => Object.values(GRAPHQL_TARGETS).includes(target);
+const isKnownGraphQlTarget = target =>
```
Comment (tdeekens) [inline packages/application-shell/src/apollo-links/header-link.js]: Just renaming I couldn't keep my fingers from.

File: packages/application-shell/src/apollo-links/header-link.js [hunk 2]
```diff
@@ -7,7 +8,9 @@ import {
...
+  Object.values(GRAPHQL_TARGETS).includes(target);
+const omitEmptyHeaders = headers => omitEmpty(headers);
```
Comment (tdeekens) [inline packages/application-shell/src/apollo-links/header-link.js]: Just to give it a "meaningful" name.

File: packages/application-shell/src/apollo-links/header-link.js [hunk 3]
```diff
@@ -43,12 +46,12 @@ const headerLink = new ApolloLink((operation, forward) => {
 
   operation.setContext({
     credentials: 'include',
-    headers: {
+    headers: omitEmptyHeaders({
```
Comment (tdeekens) [inline packages/application-shell/src/apollo-links/header-link.js]: Found this easier to follow than the "optional spreading" stuff.

File: packages/application-shell/src/apollo-links/header-link.js (hunk 2 above)
Comment (emmenko) [inline packages/application-shell/src/apollo-links/header-link.js]: Nit: is this function necessary? Why not simply using `omitEmpty` directly?

Comment (emmenko) [reply packages/application-shell/src/apollo-links/header-link.js]: > Just to give it a "meaningful" name.

To be honest, I find `headers: omitEmpty({ ... })` already quite clear. But yeah, was just curious

**Pull request merged**